include/envoy/event/dispatcher.h

```diff
@@ -104,10 +104,13 @@ class Dispatcher {
    * dispatcher.
    * @param resolvers supplies the addresses of DNS resolvers that this resolver should use. If left
    * empty, it will not use any specific resolvers, but use defaults (/etc/resolv.conf)
+   * @param use_tcp_for_dns_lookups if set to true, tcp will be used to perform dns lookups.
+   * Otherwise, udp is used.
    * @return Network::DnsResolverSharedPtr that is owned by the caller.
    */
   virtual Network::DnsResolverSharedPtr
-  createDnsResolver(const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers) PURE;
+  createDnsResolver(const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers,
+                    const bool use_tcp_for_dns_lookups) PURE;
```

Comment:
`const bool` doesn't do anything on interfaces for value types, so I'd make this just `bool`.